File: AIweather/backend/src/main/java/com/ai/weather/util/WeatherCodeMapper.java

package com.ai.weather.util;

import java.util.HashMap;
import java.util.Map;

public class WeatherCodeMapper {

    private static final Map<Integer, String> codeMap = new HashMap<>();

    static {
        codeMap.put(0, "晴朗");
        codeMap.put(1, "大部晴朗");
        codeMap.put(2, "局部多云");
        codeMap.put(3, "阴天");
        codeMap.put(45, "雾");
        codeMap.put(48, "冻雾");
        codeMap.put(51, "毛毛雨");
        codeMap.put(53, "毛毛雨");
        codeMap.put(55, "毛毛雨");
        codeMap.put(56, "冻毛毛雨");
        codeMap.put(57, "冻毛毛雨");
        codeMap.put(61, "小雨");
        codeMap.put(63, "中雨");
        codeMap.put(65, "大雨");
        codeMap.put(66, "冻雨");
        codeMap.put(67, "冻雨");
        codeMap.put(71, "小雪");
        codeMap.put(73, "中雪");
        codeMap.put(75, "大雪");
        codeMap.put(77, "雪粒");
        codeMap.put(80, "阵雨");
        codeMap.put(81, "强阵雨");
        codeMap.put(82, "暴雨");
        codeMap.put(85, "阵雪");
        codeMap.put(86, "强阵雪");
        codeMap.put(95, "雷阵雨");
        codeMap.put(96, "雷阵雨伴冰雹");
        codeMap.put(99, "强雷阵雨伴冰雹");
    }

    public static String describe(int code) {
        return codeMap.getOrDefault(code, "未知");
    }

    public enum WeatherCategory {
        SUNNY, CLOUDY, FOGGY, RAINY, SNOWY, THUNDER
    }

    public static WeatherCategory category(int code) {
        if (code == 0 || code == 1) {
            return WeatherCategory.SUNNY;
        } else if (code == 2 || code == 3) {
            return WeatherCategory.CLOUDY;
        } else if (code == 45 || code == 48) {
            return WeatherCategory.FOGGY;
        } else if ((code >= 51 && code <= 67) || (code >= 80 && code <= 82)) {
            return WeatherCategory.RAINY;
        } else if ((code >= 71 && code <= 77) || (code >= 85 && code <= 86)) {
            return WeatherCategory.SNOWY;
        } else if (code >= 95 && code <= 99) {
            return WeatherCategory.THUNDER;
        } else {
            return WeatherCategory.SUNNY;
        }
    }

    public static String uvDescription(double uv) {
        if (uv < 3) {
            return "低";
        } else if (uv < 6) {
            return "中等";
        } else if (uv < 8) {
            return "高";
        } else if (uv < 11) {
            return "很高";
        } else {
            return "极高";
        }
    }
}

File: xm/app/src/main/java/com/ai/weather/data/model/WeatherCodeMapper.kt
package com.ai.weather.data.model

/**
 * WMO天气代码映射
 * 与参考项目的if-else链不同，使用when表达式 + Map查找，更符合Kotlin风格
 */
object WeatherCodeMapper {

    private val codeMap = mapOf(
        0 to "晴朗",
        1 to "大部晴朗",
        2 to "局部多云",
        3 to "阴天",
        45 to "雾",
        48 to "冻雾",
        51 to "毛毛雨",
        53 to "毛毛雨",
        55 to "毛毛雨",
        56 to "冻毛毛雨",
        57 to "冻毛毛雨",
        61 to "小雨",
        63 to "中雨",
        65 to "大雨",
        66 to "冻雨",
        67 to "冻雨",
        71 to "小雪",
        73 to "中雪",
        75 to "大雪",
        77 to "雪粒",
        80 to "阵雨",
        81 to "强阵雨",
        82 to "暴雨",
        85 to "阵雪",
        86 to "强阵雪",
        95 to "雷阵雨",
        96 to "雷阵雨伴冰雹",
        99 to "强雷阵雨伴冰雹"
    )

    fun describe(code: Int): String = codeMap[code] ?: "未知"

    /** 天气类别分组 - 用于UI图标与主题色选择 */
    enum class WeatherCategory { SUNNY, CLOUDY, FOGGY, RAINY, SNOWY, THUNDER }

    fun category(code: Int): WeatherCategory = when (code) {
        0, 1 -> WeatherCategory.SUNNY
        2, 3 -> WeatherCategory.CLOUDY
        45, 48 -> WeatherCategory.FOGGY
        in 51..67, in 80..82 -> WeatherCategory.RAINY
        in 71..77, in 85..86 -> WeatherCategory.SNOWY
        in 95..99 -> WeatherCategory.THUNDER
        else -> WeatherCategory.SUNNY
    }

    fun uvDescription(uv: Double): String = when {
        uv < 3 -> "低"
        uv < 6 -> "中等"
        uv < 8 -> "高"
        uv < 11 -> "很高"
        else -> "极高"
    }
}
